packages/controller/main.js

```diff
@@ -844,6 +855,10 @@ function createObjects(onConnect) {
                     logger.error(`${hostLogPrefix} cannot process: ${id}: ${err} / ${err.stack}`);
                 }
             }
+        },
+        primaryHostLost: () => {
+            logger.info('The primary host is no longer active. Checking responsibilities.');
```

Comment:
I would add a guard that this is not executed if we currently shutdown this controller ...